ckan/lib/search/__init__.py

```diff
@@ -1,318 +1,23 @@
-import logging
-import sys
-import cgitb
-import warnings
-import xml.dom.minidom
-import urllib2
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+from ckan import plugins
 
-from pylons import config
-from paste.deploy.converters import asbool
 
-import ckan.model as model
-import ckan.plugins as p
-import ckan.logic as logic
+def index(package):
+    for plugin in plugins.PluginImplementations(plugins.IIndexer):
+        return plugin.index(package)
```

Comment:
Just stumbled over this and was wondering if this is the standard way to check for plugins or if there is another way? It seems a little confusing to me that we are looping over the plugins here when what we are actually doing is taking the first implementer and returning its value. Meaning that the loop implicates that we are doing something for each plugin but we already return after the first one...